src/compiler/crystal/tools/doc/type.cr

```diff
@@ -726,4 +727,25 @@ class Crystal::Doc::Type
       end
     )
   end
+
+  def to_json(io)
+    {
+      kind:             kind,
+      name:             name,
+      abstract:         abstract?,
```

Comment:
Perhaps an attribute whether the type is generic? So that nobody has to parse/guess it from `name`.